compiler/src/Language/Bond/Parser.hs

```diff
@@ -93,31 +72,38 @@ processImport :: Import -> Parser()
 processImport (Import file) = do
     Environment { currentFile = currentFile, resolveImport = resolveImport } <- ask
     (path, content) <- liftIO $ resolveImport currentFile file
-    Symbols { imports = imports } <- getState
+    Symbols { imports = imports } <- get
     if path `elem` imports then return () else do
-            modifyState (\u -> u { imports = path:imports } )
+            modify (\u -> u { imports = path:imports } )
             setInput content
             setPosition $ initialPos path
             void $ local (\e -> e { currentFile = path }) bond
 
 -- parser for struct, enum or type alias declaration/definition
 declaration :: Parser Declaration
 declaration = do
+    -- When adding a new Declaration parser, order matters in the following command.
+    -- Parsers must fail to consume ANY token for the next parser to be able to succesfully work
+    -- unless the parser is encapsulated in a try statement. For more info on try and <|> see:
+    -- https://hackage.haskell.org/package/megaparsec-6.2.0/docs/Text-Megaparsec.html#v:try
     decl <- try forward
```

Comment:
We want to avoid these try statements that encompass whole parsers, this will avoid backtracking and losing information for better error messages. Try statements should be used when parsing 'branches' or where the syntax diverges. This way if when we successfully parse the token 'service', and then later we hit a parsing error we don't backtrack all the way to here, instead just fail iniside the service parser. This is trickier with struct and struct view that share some common syntax.

Reply:
Are you planning to fold forward into structDeclaration?

Reply:
No since it would be the same code. If I move it to structDecleration

```
forward = Forward <$> asks currentNamespaces <*> name <*> parameters <* semi <?> "forward declaration"
  where
name = keyword "struct" *> identifier
```
would end up sharing everything except the ```semi``` which would end up being a ```try semi``` and no benefit would be achieved since there is nothing else to parse after that. After that try statement we would end up running the other parsers all the time since if its not a semi it will likely be a struct or a view of.

Reply:
Plus forwards don't have attributes and it would complicate the code for no benefit at all.